code/modules/modular_computers/file_system/programs/generic/file_manager.dm

```diff
@@ -325,6 +325,20 @@
 			current_transfer = new(current_disk, dst, dst_dir, F, copying == "Copy" ? TRUE : FALSE)
 			ui_header = "downloader_running.gif"
 
+	if(href_list["PRG_runfile"])
+		. = TOPIC_HANDLED
+		if(!open_file)
+			return
+		var/datum/computer_file/program/P = current_disk.get_file(open_file, current_directory)
+		if(istype(P))
+			computer.run_program(P.filename)
+			return TOPIC_HANDLED
+		var/datum/computer_file/data/F = current_disk.get_file(open_file, current_directory)
```

Comment:
These point to the same file, we're just checking a different type.
```suggestion
		var/datum/computer_file/data/F = P
```
Another alternative could be:
```dm
		var/datum/computer_file/runnable_file = current_disk.get_file(open_file, current_directory)
		if(istype(runnable_file, /datum/computer_file/program))
			computer.run_program(runnable_file.filename)
			return TOPIC_HANDLED
		if(!runnable_file || !istype(runnable_file, /datum/computer_file/data))
			return TOPIC_NOACTION
		computer.run_script(usr, runnable_file)
		return TOPIC_HANDLED
```
Since we're not using any methods defined on those subtypes, we can avoid the cast altogether and just check the type explicitly.

Reply:
Actually, I'll PR this myself with some of my other OS/filesystem changes, probably.